src/vipersci/vis/pds/create_raw.py

```diff
@@ -0,0 +1,403 @@
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Reuse is permitted under the terms of the license.
+# The AUTHORS file and the LICENSE file are at the
+# top level of this library.
+
+import argparse
+from datetime import datetime, date, timezone
+import hashlib
+from importlib import resources
+import io
+import json
+import logging
+import sys
+from typing import Union
+from pathlib import Path
+from warnings import warn
+
+from genshi.template import MarkupTemplate
+import numpy as np
+import numpy.typing as npt
+from skimage.io import imread, imsave  # maybe just imageio here?
+from sqlalchemy import create_engine
+from sqlalchemy.orm import Session, sessionmaker
+from tifftools import read_tiff
+
+import vipersci
+from vipersci.vis.db.raw_products import RawProduct
+from vipersci.pds import pid as pds
+from vipersci import util
+
+logger = logging.getLogger(__name__)
+
+
+def arg_parser():
+    parser = argparse.ArgumentParser(
+        description=__doc__,
+        parents=[util.parent_parser()]
+    )
+    parser.add_argument(
+        "-d", "--dburl",
+        help="Database with a raw_products table which will be written to. "
+             "If not given, no database will be written to.  Example: "
+             "postgresql://postgres:NotTheDefault@localhost/visdb"
+    )
+    parser.add_argument(
+        "--json",
+        type=Path,
+        required=True,
+        help="JSON file, with raw product meta-data."
+    )
+    parser.add_argument(
+        "-t", "--template",
+        type=Path,
+        help="Genshi XML file template.  Will default to the raw-template.xml "
+             "file distributed with the module."
+    )
+    parser.add_argument(
+        "--image",
+        type=Path,
+        help="Optional path to source image file which will be read and "
+             "converted into a TIFF product.  If no path is given, then "
+             "Only a .xml label will be written and no image."
+    )
+    parser.add_argument(
+        "--tiff",
+        type=Path,
+        help="Optional pre-existing TIFF file.  This file will be inspected "
+             "and an .xml label will be written."
+    )
+    parser.add_argument(
+        "-o", "--output_dir",
+        type=Path,
+        default=Path.cwd(),
+        help="Output directory for label. Defaults to current working directory."
+    )
+    return parser
+
+
+def main():
+    args = arg_parser().parse_args()
+    util.set_logger(args.verbose)
+
+    with open(args.json) as f:
+        jsondata = json.load(f)
+
+    # I'm not sure where these are coming from, let's hard-code them for now:
+    metadata = {
+        "mission_phase": "TEST",
+        "purpose": "Navigation",
+        "onboard_compression_ratio": 64
+    }
+
+    # This allows values in jsondata to override the hard-coded values above.
+    metadata.update(jsondata)
+
+    if args.image is not None:
+        image = imread(str(args.image))
```

Comment:
There's an implicit (but not communicated) priority here that --image overrides --tiff. Should we warn the user if both --image and --tiff are specified in the arguments to this program?

Reply:
Sure, will add a note to the `--tiff` help string.